include/aos/cm/networkmanager.hpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2025 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_CM_NETWORKMANAGER_HPP_
+#define AOS_CM_NETWORKMANAGER_HPP_
+
+#include <aos/common/types.hpp>
+
+namespace aos::cm::networkmanager {
+
+/** @addtogroup CM NetworkManager
+ *  @{
+ */
+
+/**
+ * Max number of allowed connections.
+ */
+static constexpr auto cMaxNumConnections = AOS_CONFIG_NETWORKMANAGER_CONNECTIONS_PER_INSTANCE_MAX_COUNT;
+
+/**
+ * Max length of connection name.
+ */
+static constexpr auto cConnectionNameLen = AOS_CONFIG_NETWORKMANAGER_CONNECTION_NAME_LEN;
+
+/**
+ * Represents network configuration parameters for a service instance.
+ */
+struct NetworkParameters {
+    /**
+     * List of hostnames.
+     */
+    StaticArray<StaticString<cHostNameLen>, cMaxNumHosts> mHosts;
+
+    /**
+     * List of allowed service connections.
+     */
+    StaticArray<StaticString<cConnectionNameLen>, cMaxNumConnections> mAllowedConnections;
+
+    /**
+     * List of ports exposed by service instance.
+     */
+    StaticArray<StaticString<cExposedPortLen>, cMaxNumExposedPorts> mExposedPorts;
+};
+
+/**
+ * Interface for managing network parameters for service instances.
+ */
+class NetworkManagerItf {
+public:
+    /**
+     * Destructor.
+     */
+    virtual ~NetworkManagerItf() = default;
+
+    /**
+     * Prepares and assigns network parameters for a service instance.
+     *
+     * @param instanceID instance identifier.
+     * @param networkID identifier of the target network.
+     * @param params input parameters for network setup.
+     * @param[out] result result network parameters.
+     * @return Error.
+     */
+    virtual Error PrepareInstanceNetworkParameters(const InstanceIdent& instanceID, const String& networkID,
+        const NetworkParameters& params, NetworkParameters& result)
+        = 0;
+
+    /**
+     * Removes assigned network parameters for the specified service instance.
+     *
+     * @param instanceID instance identifier.
+     * @return Error.
+     */
+    virtual Error RemoveInstanceNetworkParameters(const InstanceIdent& instanceID) = 0;
+
+    /**
+     * Restarts DNS server.
+     *
+     * @return Error.
+     */
+    virtual Error RestartDNSServer() = 0;
+
+    /**
+     * Returns all service instances registered in network manager.
+     *
+     * @return const Array<InstanceIdent>&.
+     */
+    virtual const Array<InstanceIdent>& GetInstances() = 0;
```

Comment:
i think better to pass array as out parameter than return array
`Error GetInstances(Array<InstanceIdent>&) const = 0;`